pkg/storage/backend-storage/backend-storage.go

```diff
@@ -104,6 +111,119 @@ func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachin
 	return pvcForMigrationTargetFromStore(pvcStore, migration)
 }
 
+func RecoverFromBrokenMigration(client kubecli.KubevirtClient, migrationIndexer cache.Indexer, pvcStore cache.Store, vmi *corev1.VirtualMachineInstance, launcherImage string) (*v1.PersistentVolumeClaim, error, bool) {
+	var pvc *v1.PersistentVolumeClaim
+
+	migration, err := migrations.MigrationForVMI(migrationIndexer, vmi)
+	if err != nil || migration == nil {
+		return nil, err, false
+	}
+	if migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == migration.Status.MigrationState.SourcePersistentStatePVCName {
+		// The migration either didn't actually start, or the backend storage is RWX. Either way we're good, we can delete it.
+		err := client.VirtualMachineInstanceMigration(migration.Namespace).Delete(context.Background(), migration.Name, metav1.DeleteOptions{})
+		return nil, err, false
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "backend-storage-recover-",
+		},
+		Spec: batchv1.JobSpec{
+			ActiveDeadlineSeconds:   pointer.P(int64(30)),
+			BackoffLimit:            pointer.P(int32(1)),
+			TTLSecondsAfterFinished: pointer.P(int32(30)),
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "backend-storage-recover-",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					SecurityContext: &v1.PodSecurityContext{
+						RunAsNonRoot: pointer.P(true),
+						RunAsUser:    pointer.P(int64(util.NonRootUID)),
+						RunAsGroup:   pointer.P(int64(util.NonRootUID)),
+						FSGroup:      pointer.P(int64(util.NonRootUID)),
+						SeccompProfile: &v1.SeccompProfile{
+							Type: v1.SeccompProfileTypeRuntimeDefault,
+						},
+					},
+					Containers: []v1.Container{{
+						Name: "container",
+						SecurityContext: &v1.SecurityContext{
+							AllowPrivilegeEscalation: pointer.P(false),
+							Capabilities:             &v1.Capabilities{Drop: []v1.Capability{"ALL"}},
+						},
+						Image:   launcherImage,
+						Command: []string{"ls"},
+						Args:    []string{"/meta/migrated"},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "backend-storage",
+							MountPath: "/meta",
+							SubPath:   "meta",
+						}},
+					}},
+					Volumes: []v1.Volume{{
+						Name: "backend-storage",
+						VolumeSource: v1.VolumeSource{
+							PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+								ClaimName: migration.Status.MigrationState.SourcePersistentStatePVCName,
```

Comment:
Might be this volume be owned by a launcher pod? I think there are certain storage classes which avoid a PVC to be used by multiple pods at the same time if the mode is RWO. @awels please correct me here if I'm wrong.

Reply:
This is called by the VMI controller before starting a VMI, so no there shouldn't be any virt-launcher pod with this volume at this point.